pkg/sql/plan/flatten_subquery.go

```diff
@@ -345,6 +349,27 @@ func (builder *QueryBuilder) generateComparison(op string, child *plan.Expr, ctx
 	}
 }
 
+func (builder *QueryBuilder) findAggFunc(nodeID int32) bool {
```

Comment:
Don't traverse the sub-tree to find aggregates. It may report false positive for a legal sub-sub-query.

`len(subCtx.aggregates) > 0` should work here.

Reply:
fixed